### ISSUE TITLE
[Feature Request]: Get external mods when Winget-Install.ps1 is called

### ISSUE THREAD

**Issue by KreativeKrise** [Romanitho/Winget-AutoUpdate#794]
### The request

External mods should not only be installed when Winget-Upgrade.ps1 is called, but also when Winget-Install.ps1 is called.

**Comment by AndrewDemski-ad-gmail-com** [comment]
Am I the only one reading it as if ($A -and $A)?
________________________________
From: KreativeKrise ***@***.***>
Sent: 16 December 2024 10:14
To: Romanitho/Winget-AutoUpdate ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Romanitho/Winget-AutoUpdate] [Feature Request]: Get external mods when Winget-Install.ps1 is called (Issue #794)


The request

External mods should not only be installed when Winget-Upgrade.ps1 is called, but also when Winget-Upgrade.ps1 is called.

Is your feature request related to a problem?

No response

Additional information

No response

**Comment by KreativeKrise** [comment]
> Am I the only one reading it as if ($A -and $A)?

Sorry, it's a copy & paste failure. I edited it.

**Comment by KnifMelti** [comment]
> External mods should not only be installed when Winget-Upgrade.ps1 is called, but also when Winget-Install.ps1 is called.

They are already used in `Winget-Install.ps1`:
https://github.com/Romanitho/Winget-AutoUpdate/blob/f4bcca43c3f14b186bfc4a5ce9121e67759c12a9/Sources/Winget-AutoUpdate/Winget-Install.ps1#L146

**Comment by KreativeKrise** [comment]
Yes, they are used if they are available locally. But the mods are not fetched from the external source before this install starts. You currently have to run Winget-Upgrade.ps1 in advance so that the mods are copied locally from the external source.

**Comment by KnifMelti** [comment]
Ah, so that's what you ment with _External mods should not only be **installed**_ 
My bad...

**Comment by Romanitho** [comment]
Maybe we can add a `Get-Mods.ps1` function with this content that can be reused in `Winget-Install.ps1` ?
https://github.com/Romanitho/Winget-AutoUpdate/blob/f4bcca43c3f14b186bfc4a5ce9121e67759c12a9/Sources/Winget-AutoUpdate/Winget-Upgrade.ps1#L162-L203

**Comment by KnifMelti** [comment]
> Maybe we can add a `Get-Mods.ps1` function with this content that can be reused in `Winget-Install.ps1` ?

Shouldn't it be:
https://github.com/Romanitho/Winget-AutoUpdate/blob/f4bcca43c3f14b186bfc4a5ce9121e67759c12a9/Sources/Winget-AutoUpdate/Winget-Upgrade.ps1#L206-L234

**Comment by Romanitho** [comment]
probably 😅
